contracts/FeralfileArtworkV4.sol

```diff
@@ -1,112 +1,602 @@
+        address from,
+        uint256 tokenId
+    ) private {
+        // To prevent a gap in from's tokens array, we store the last token in the index of the token to delete, and
+        // then delete the last slot (swap and pop).
+        uint256 lastTokenIndex = ERC721.balanceOf(from) - 1;
+        uint256 tokenIndex = _ownedTokensIndex[tokenId];
+
+        // When the token to delete is the last token, the swap operation is unnecessary
+        if (tokenIndex != lastTokenIndex) {
+            uint256 lastTokenId = _ownedTokens[from][lastTokenIndex];
+
+            _ownedTokens[from][tokenIndex] = lastTokenId; // Move the last token to the slot of the to-delete token
+            _ownedTokensIndex[lastTokenId] = tokenIndex; // Update the moved token's index
+        }
+
+        delete _ownedTokensIndex[tokenId];
+        _ownedTokens[from].pop();
+    }
+
+    /// @dev Modify from ERC721Enumerable
+    function _addTokenToOwnerEnumeration(address to, uint256 tokenId) private {
+        uint256[] storage tokens = _ownedTokens[to];
+        uint256 length = tokens.length;
+        tokens.push(tokenId);
+        _ownedTokensIndex[tokenId] = length;
+    }
+
+    /// @notice Mint new collection of Artwork
+    /// @dev the function iterates over the array of MintData to call the internal function _mintArtwork
+    /// @param data an array of MintData
+    function mintArtworks(
+        MintData[] calldata data
+    ) external virtual onlyAuthorized {
+        for (uint256 i = 0; i < data.length; i++) {
+            _mintArtwork(data[i].seriesId, data[i].tokenId, data[i].owner);
+        }
+    }
+
+    function _mintArtwork(
+        uint256 seriesId,
+        uint256 tokenId,
+        address owner
+    ) internal {
+        // pre-condition checks
+        require(
+            mintable,
+            "FeralfileExhibitionV4: contract doesn't allow to mint"
+        );
+        require(
+            _seriesExists(seriesId),
+            string(
+                abi.encodePacked(
+                    "FeralfileExhibitionV4: seriesId doesn't exist: ",
+                    Strings.toString(seriesId)
+                )
+            )
+        );
+
+        // mint
+        totalSupply += 1;
+        _seriesTotalSupplies[seriesId] += 1;
+        _allArtworks[tokenId] = Artwork(seriesId, tokenId);
+        _mint(owner, tokenId);
+        require(
+            _seriesTotalSupplies[seriesId] <= _seriesMaxSupplies[seriesId],
+            "FeralfileExhibitionV4: no slots available"
+        );
+
+        // emit event
+        emit NewArtwork(owner, seriesId, tokenId);
+    }
+
+    /// @notice Burn a collection of artworks
+    /// @dev the function iterates over the array of token ID to call the internal function _burnArtwork
+    /// @param tokenIds an array of token ID
+    function burnArtworks(uint256[] memory tokenIds) external {
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            _burnArtwork(tokenIds[i]);
```

Comment:
```suggestion
            require(_isApprovedOrOwner(_msgSender(), tokenId), "ERC721: caller is not token owner or approved");
            _burnArtwork(tokenIds[i]);
```
@jollyjoker992 the external function can be called for users so we need to check the ownership before burn tokens.